grayskull/pypi/pypi.py

```diff
@@ -787,15 +809,23 @@ def _extract_requirements(self, metadata: dict) -> dict:
         :return: all requirement section
         """
         name = metadata["name"]
-        requires_dist = PyPi._format_dependencies(metadata.get("requires_dist"), name)
-        setup_requires = (
-            metadata.get("setup_requires") if metadata.get("setup_requires") else []
-        )
-        host_req = PyPi._format_dependencies(setup_requires, name)
+        if metadata.get("requires_dist"):
+            requires_dist = PyPi._format_dependencies(metadata.get("requires_dist"), name)
+        else:
+            requires_dist = metadata.get("requires_dist")
+
+        if metadata.get("setup_requires"):
+            setup_requires = (
+                metadata.get("setup_requires") if metadata.get("setup_requires") else []
+            )
+            host_req = PyPi._format_dependencies(setup_requires, name)
+        else:
```

Comment:
This clause makes no sense :( We are already testing for the presence of the key implicitly above.